File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        hashset = set()

        left = right = 0 
        
        while right < len(nums):
            if nums[right] not in hashset:
                hashset.add(nums[right])
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
            right += 1           
        
        return left

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int l = 0, n = nums.size();

        for (int r = 1; r < n; r++){
            if (nums[r] != nums[r-1]){
                nums[++l] = nums[r];
            }
        }

        return l + 1;
    }
};